docs/index.rst

```diff
@@ -1406,6 +1445,76 @@ It is possible to use factories in
     For the same reason as above, it is not possible to use `Factories as Services`_ with required
     constructor arguments (the container is not yet available).
 
+.. tip::
+
+    ``ModelFactory::new()->many()`` and ``ModelFactory::new()->sequence()`` return a special ``FactoryCollection`` object
+    which can be used to generate data providers:
+
+    .. code-block:: php
+
+        use App\Factory\PostFactory;
+
+        /**
+         * @dataProvider postDataProvider
+         */
+        public function test_post_via_data_provider(PostFactory $factory): void
+        {
+            $factory->create();
+
+            // ...
+        }
+
+        public static function postDataProvider(): iterable
+        {
+            $sequence = PostFactory::new()->sequence(
+                [
+                    ['title' => 'foo'],
+                    ['title' => 'bar'],
+                ]
+            );
+
+            foreach ($sequence as $factory) {
+                yield [$factory];
+            }
```

Comment:
How common do you think this pattern would be? Just returning a single factory (wrapped in an array) from a data provider? I think the `dataProvider()` method we had before was confusing because of the extra parameters it takes, but what about:

```php
            yield from PostFactory::new()->sequence(
                [
                    ['title' => 'foo'],
                    ['title' => 'bar'],
                ]
            )->asDataProvider();
```

As a shortcut to this block?

Reply:
but I can definitely introduce the `asDataProvider()` method (I prefer this new name, btw)

Reply:
Ok, let's do this.

Reply:
done :heavy_check_mark:

Reply:
done :heavy_check_mark: